### PR TITLE
OPRUN-3570: Downstream testing of OpenShift Max Version property

### PR DESCRIPTION
**Pull request by gallettilance**

### CODE REVIEW

File: test/extended/testdata/olmv1/install-operator.yaml
Comment (tmshort) [inline test/extended/testdata/olmv1/install-operator.yaml:1]: Because you're modifying this file, you need to run `make update` to update the file inside of the executable.

File: test/extended/cluster-olm-operator/cluster-olm-operator.go
Comment (tmshort) [inline test/extended/cluster-olm-operator/cluster-olm-operator.go:49]: This is a dependency on the order of the status fields. Are we comfortable with that?

File: test/extended/cluster-olm-operator/cluster-olm-operator.go
Comment (tmshort) [inline test/extended/cluster-olm-operator/cluster-olm-operator.go:71]: I'm not seeing a `delete` operation here (no cleanup). Which might require an `AfterEach` clause.

File: test/extended/olm/olmv1.go
Comment (tmshort) [inline test/extended/olm/olmv1.go:187]: Can the `return` statemment and following `})` be put on their own lines for readability?

File: test/extended/olm/olmv1.go
Comment (tmshort) [inline test/extended/olm/olmv1.go:150]: Can the `return` statemment and following `})` be put on their own lines for readability?

File: test/extended/olm/olmv1.go
Comment (tmshort) [inline test/extended/olm/olmv1.go:205]: I would this this function would actually do the read and write, based on its name.
The return value would be the new name of the file, plus an error...

File: test/extended/util/annotate/generated/zz_generated.annotations.go
Comment (tmshort) [inline test/extended/util/annotate/generated/zz_generated.annotations.go:1686]: This is a bit concerning to me... as it's implying that these could be running in parallel... so you might be trying to run your two tests _at the same time_, which would give unpredictable results. You may need to combine the tests into a single `g.It()` clause.

File: test/extended/olm/olmv1.go
Comment (tmshort) [reply test/extended/olm/olmv1.go:205]: Maybe just a rename to something like `updateCe()`?

**Review by m1kola — COMMENTED** [review]
Looks reasonable to me. I think we should be good after addressing Todd's comments (the one about parallel execution particularly).

File: test/extended/util/annotate/generated/zz_generated.annotations.go
Comment (tmshort) [reply test/extended/util/annotate/generated/zz_generated.annotations.go:1686]: If that avoids the parallelism, then that's ok.

**Comment by tmshort** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview-serial

^^^ This should run some tech preview jobs (this was done in my #29249), which will test that the test actually run (and don't skip)

**Comment by tmshort** [comment]
Didn't mean to run this twice... Don't edit comments with the /payload-job trigger!

File: test/extended/util/annotate/generated/zz_generated.annotations.go
Comment (m1kola) [reply test/extended/util/annotate/generated/zz_generated.annotations.go:1686]: I see it in other sequential jobs.

From https://onsi.github.io/ginkgo/#ordered-containers:

> Specs in an Ordered container may run in parallel with respect to other specs, but they will always run sequentially on the same parallel process.


Should be good enough for this test I think.

**Comment by jianzhangbjz** [comment]
cc: @Xia-Zhao-rh ^^

**Comment by tmshort** [comment]
It does appear that the `sig-olmv1` tests failed: https://prow.ci.openshift.org/view/gs/test-platform-results/logs/openshift-origin-29242-ci-4.18-e2e-aws-ovn-techpreview/1854190700078305280

**Comment by tmshort** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview-serial

**Review by tmshort — APPROVED** [review]
/lgtm
We should wait for those other tests to pass, and we'll need more approvals

**Comment by dgoodwin** [comment]
/hold

Why did the last payload job attempt fail both jobs? https://github.com/openshift/origin/pull/29242#issuecomment-2463180053

**Comment by tmshort** [comment]
Yeah, it looks as though the tests are still not working in parallel @gallettilance

File: test/extended/testdata/olmv1/install-operator.yaml
Comment (tmshort) [inline test/extended/testdata/olmv1/install-operator.yaml:32]: If you go through another iteration, perhaps replace `{REPLACE}` with `{NAMESPACE}`?

**Comment by tmshort** [comment]
/retest

None of the failures seem to be due to [sig-olmv1]

**Comment by tmshort** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview-serial

**Comment by gallettilance** [comment]
/test ci/prow/e2e-aws-ovn-kube-apiserver-rollout ci/prow/e2e-aws-ovn-single-node-serial 
ci/prow/e2e-metal-ipi-ovn-kube-apiserver-rollout

**Comment by tmshort** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview-serial

This one had failed, but `periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview` passed

**Comment by tmshort** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-techpreview-serial

**Pull request merged**
